R/util.R

```diff
@@ -14,7 +6,7 @@ simplify_terms <- function(x) {
   is_terms <- inherits(x, "terms")
 
   if (!is_terms) {
-    abort("`x` must be a terms object")
+    cli::cli_abort("{.arg x} must be a {.cls terms} object.")
```

Comment:
More and more, I"m partial to adding a `", not {.obj_type_friendly {x}}."` so that there is a clue as to what went wrong.

Reply:
You probably want a custom `check_terms()` then, which would call `stop_input_type()` from the standalone type checker file. That automatically adds the `", not {.obj_type_friendly {x}}."` bit for you and uses a standardized message

Reply:
This is the general vibe of a custom checker:

```
> ivs:::check_data_frame
function (x, ..., allow_null = FALSE, arg = caller_arg(x), call = caller_env()) 
{
    if (!missing(x)) {
        if (is.data.frame(x)) {
            return(invisible(NULL))
        }
        if (allow_null && is_null(x)) {
            return(invisible(NULL))
        }
    }
    stop_input_type(x, "a data frame", ..., allow_null = allow_null, 
        arg = arg, call = call)
}
```

<img width="375" alt="Screenshot 2024-10-24 at 2 30 00 PM" src="https://github.com/user-attachments/assets/24c32047-edb9-417e-86cb-f00b3ed848a9">